scripts/google/covid_mobility/covidmobility.py

```diff
@@ -0,0 +1,233 @@
+
+        for row in rows:
+            # If flag less_ouput is enabled, then only output less than 1% of the data.
+            if less_output and skip_value(skip_pct=0.99995):
+                continue
+
+            row_as_list: list = row.split(',')
+
+            # There must be 11 rows, otherwise skip that row.
+            if len(row_as_list) != 11:
+                continue
+
+            try:
+                row_data: dict = csv_row_to_obj(row_as_list)
+            # If it's not a US sub_region, skip it.
+            except KeyError:
+                print('Skipped: ' + str(row_as_list))
+                continue
+
+            data.append(row_data)
+
+            dcid: str = row_data['dcid']
+            node_ids = generate_places_node_ids_for_row(row_data)
+
+            mobility_locations[dcid] = {"place_categories": node_ids}
+
+    write_statistical_pop_node(mobility_locations, output_path)
+    write_observation_nodes(data, mobility_locations, output_path)
+
+def csv_row_to_obj(row: list) -> dict:
+    """
+    Given a row from an input file, generate an object for it.
+    :arg row: a list of strings separated by commas, where each index is a column.
+    :returns data_as_dict: the row's data where each column is a key.
+    :except KeyError if there is no dcid for a given place.
+    """
+
+    # Store values as variables
+    country_code: str = row[0] # ES, US are example of country codes.
+    country: str = row[1].replace(' ', '') # util.COUNTRY_MAP requires no white-spaces
+    sub_region_1: str = row[2].replace(' ', '') # util.USSTATE_MAP requires no white-spaces
+    sub_region_2: str = row[3] # Whitespaces are okay here
+    dcid: str = get_dcid_from_region(sub_region_2, sub_region_1, country_code) # country/ESP is an example
+
+    # Return a dictionary of the row
+    return {
+        'dcid': dcid,
+        'country_code': country_code,
+        'country': country,
+        'sub_region_1': sub_region_1,
+        'sub_region_2': sub_region_2,
+        'date': row[4],
+        'LocalBusiness': row[5],
+        'GroceryStore&Pharmacy': row[6],
+        'Park': row[7],
+        'TransportHub': row[8],
+        'Workplace': row[9],
+        'Residence': row[10]
+    }
+
+def generate_places_node_ids_for_row(row_data: dict) -> dict:
+    """Given row data, return the node local ids for each place
+
+    Args:
+        row_data (dict): row_data in the form of an object
+
+    Returns:
+        node_ids: object containing place -> node Id
+    """
+    location_name: str = row_data['sub_region_2'] + row_data['sub_region_1'] + row_data['country']
+    # Ignore any non-ascii characters
+    location_name: str = ''.join([i if ord(i) < 128 else '_' for i in location_name])
+    # Get rid of any whitespaces
+    location_name: str = location_name.replace(" ", "")
+    location_dcid: str = row_data['dcid']
+
+    return {
+        "LocalBusiness": f"{location_name}LocalBusiness",
```

Comment:
WDYT of trying something like 

return """
triple
triple
triple
"""

Or using https://docs.python.org/3.1/library/textwrap.html ?

Reply:
I don't know if this applies anymore. Let me know.